notes/error_iterator.md

```diff
@@ -0,0 +1,63 @@
+# pulse
```

Comment:
Notes like this should probably be kept to issues / PR comments, rather than committing files like this to the repo

Reply:
I think I lost a .gitignore somewhere along the way